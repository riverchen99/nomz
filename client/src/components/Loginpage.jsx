import React from 'react';
import { Link } from 'react-router-dom';
import Button from './Button';
import { StyledLoginpage, Title } from './StyledLoginpage';

class Loginpage extends React.Component {
  render () {
    return (
      <StyledLoginpage>
        <Title>nomz</Title>
        <p>Personalized recommendations based on your ratings</p>
        <Button text="Login" color="white"/>
        <p>Or <Link to="/recommend">continue as guest</Link></p>
      </StyledLoginpage>
    )
  }
}

export default Loginpage;