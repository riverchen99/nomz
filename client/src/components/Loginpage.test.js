import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';
import Loginpage from './Loginpage';

configure({ adapter: new Adapter() });

describe('Login page component', () => {
  it('should render correctly', () => {
    const component = shallow(<MemoryRouter><Loginpage /></MemoryRouter>);
    expect(component).toMatchSnapshot();
  });
  // this test is only valid for mvp stage since auth isn't final
  it('has correct Link path to recommendations', () => {
    const component = mount(<MemoryRouter><Loginpage /></MemoryRouter>);
    expect(component.find('Link').props().to).toEqual('/recommend');
    expect(component.find('Link')).toMatchSnapshot();
  });
});
