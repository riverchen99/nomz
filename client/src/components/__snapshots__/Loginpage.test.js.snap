// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page component has correct Link path to recommendations 1`] = `ReactWrapper {}`;

exports[`Login page component should render correctly 1`] = `ShallowWrapper {}`;
